### ISSUE TITLE
New rule: no-global-regex-flag-to-query

### ISSUE THREAD

**Issue by timdeschryver** [testing-library/eslint-plugin-testing-library#559]
### Name for new rule

no-global-regex-flag-to-query

### Description of the new rule

The rule would warn when a user uses the global regex flag to search for text/name.
A regex that uses the global flag holds the `lastIndex` state when it found a match, due to this it could be that a new match can't be found.

See https://github.com/testing-library/dom-testing-library/pull/1116 for the fix within DTL, where we added a log when we detect this.

### Testing Library feature

Queries that use a regex

### Testing Library framework(s)

All

### What category of rule is this?

Suggests an alternate way of doing something

### Code examples

```ts
screen.getAllByText(/hello world/g)
screen.findByRole('button', {name: /log in/ig})
```

### Do you want to submit a pull request to make the new rule?

Yes

**Comment by Belco90** [comment]
Hey @timdeschryver! This is a great idea, I read your explanation on Twitter the other day and was a TIL for me 🤯.

Let me know if you need help implementing this!

**Comment by Belco90** [comment]
Btw, if this has to be applied to every Testing Library util, you could just select all `CallExpression` and check if they are related to Testing Library with `isTestingLibraryUtil`, and then, if so, check the args given. Just an idea!
